lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -538,6 +543,28 @@ func makeSubscriberStateHandler(desiredState string) echo.HandlerFunc {
 	}
 }
 
+func getStateExactIMSIPrefix(networkID string, typeFilter []string, keyFilter []string, keyPrefix *string, serdes serde.Registry) (state_types.StatesByID, error) {
+	// post-filter functionality to only return states with the exact IMSI prefix as specified
+	states, err := state.SearchStates(networkID, typeFilter, keyFilter, keyPrefix, serdes)
+	if err != nil {
+		return nil, err
+	}
+
+	if !funk.IsEmpty(keyPrefix) {
+		for stateID := range states {
+			matches := imsiKeyRe.FindStringSubmatch(stateID.DeviceID)
```

Comment:
This is making an assumption about the format of the keys, which happens to be true generally, but isn't guaranteed -- the format of the keys is specified on a per-state-type basis. So you'll need to follow the pattern elsewhere in this file of checking the state's type for e.g. mobilityd type